src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -782,12 +782,14 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             var elementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Count);
             var elementLocations = ArrayBuilder<Location>.GetInstance(arguments.Count);
             ArrayBuilder<string> elementNames = null;
+            ArrayBuilder<string> inferredElementNames = null;
 
             // prepare and check element names and types
             for (int i = 0; i < numElements; i++)
```

Comment:
@AlekseyTs @cston @gafter @VSadov What would you think if I more aggressively factored code of this loop? 
I would need to loop over the tuple elements twice: once for tuple element names (this code would be shared with the one I have in error reporting case), and the other to collect types and locations.